testsupport/space/spacerequest.go

```diff
@@ -141,9 +142,10 @@ func newKubeClientFromSecret(t *testing.T, cl client.Client, secretName, secretN
 	return namespaceAccessClient, adminSecret
 }
 
-// validateKubeClient validates the the kube client can access the given namespace
-func validateKubeClient(t *testing.T, namespaceAccessClient client.Client, namespace string) {
-	secretsList := &corev1.SecretList{}
-	require.NoError(t, namespaceAccessClient.List(context.TODO(), secretsList, client.InNamespace(namespace)))
-	require.NotEmpty(t, secretsList)
+// ValidateKubeClient validates the the kube client can access the given namespace by listing objects using the provided list instance
+// The list is checked to not be empty.
+func ValidateKubeClient(t *testing.T, namespaceAccessClient client.Client, namespace string, list client.ObjectList) {
```

Comment:
minor - maybe we can move this to the `testsupport/util` or `testsupport/auth` package ? Probably it will look nicer then `space.ValidateKubeClient` from a client perspective. But feel free to leave it as is.

Reply:
this is a temporal check for now - just for the migration time. so, it's fine to keep it where it is

Reply:
why is temporal ? aren't we switching to kubeconfig so we will still need to validate it in the toolchaincluster secret even later ? but maybe I'm missing something.

Reply:
It will be validated a million times in other places when it is actually used for the connections among the clusters.

But you're right, a quick explicit test for it may reduce the head-scratching when problems with the connections arise.

Reply:
if there is gonna be a problem with the kubeconfig created for ToolchainCluster, then the ToolchainCluster won't get ready in the first place (which means that we wouldn't be able to do almost anything), which means that we wouldn't even reach to this point when we could try to test it explicitly.

So as @metlos mentioned, the validity of the kubeconfig will be verified by using it, this is not the case for now - the kubeconfigs are created only (as part of the migration process) but not used yet.

Reply:
I moved the functions to `testsupport/util`.